src/hotspot/share/gc/g1/g1Policy.cpp

```diff
@@ -901,7 +901,10 @@ void G1Policy::record_collection_pause_end(double pause_time_ms) {
   assert(!(is_concurrent_start_pause(this_pause) && collector_state()->mark_or_rebuild_in_progress()),
          "If the last pause has been concurrent start, we should not have been in the marking window");
   if (is_concurrent_start_pause(this_pause)) {
-    collector_state()->set_mark_or_rebuild_in_progress(true);
+    if (!start_concurrent_mark_cycle) {
+      abort_time_to_mixed_tracking();
+    }
+    collector_state()->set_mark_or_rebuild_in_progress(start_concurrent_mark_cycle);
```

Comment:
As mentioned, I would prefer to check the `collector_state()` here rather than a passed in value.